DSpace/dspace-angular
# Filter queries using correct operators

## References
- Fixes #915
- Fixes #368

## Description
This PR ensures operators are added when sending requests containing facets or filters through discovery. The operators are retrieved from the relevant part of the search link returned by the server, rather than have the UI construct the query on its own.

## Instructions for Reviewers
Changes made:
- A util file was added called `search.utils.ts` containing a method (`getFacetValueForType()`) to retrieve the relevant facet value _with_ operator from the search link returned by the server. The util file also contains functions to strip an operator from a filter's value or add one to it.
- When a user types or selects a suggested filter, the operator "query" is added to the value
- Modified labels to strip the operator (using the util function) wherever the filter values displayed (below the search box for example)

How to test:
- Visit the discovery page
- Add filters by selecting one from the sidebar or typing them yourself (and/or selecting from the suggestion dropdown)
- Verify the requests sent contain the correct operators for each filter value
- Verify any values displayed in the UI don't contain the operators

**Known Issue**
Issue [#3057](https://github.com/DSpace/DSpace/issues/3057) on the REST API might cause unexpected behaviour.

## Checklist
_This checklist provides a reminder of what we are going to look for when reviewing your PR. You need not complete this checklist prior to creating your PR (draft PRs are always welcome). If you are unsure about an item in the checklist, don't hesitate to ask. We're here to help!_
